tests/admin/proxy/ClientTableTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace admintests\admin\proxy;
+
+use admintests\AdminTestCase;
+use admintests\data\mocks\proxy\ClientTableMock;
+use luya\admin\proxy\ClientBuild;
+use luya\admin\proxy\ClientTable;
+use luya\admin\commands\ProxyController;
+
+class ClientTableTest extends AdminTestCase
+{
+    /**
+     * @throws \yii\db\Exception
+     * @expectedException \yii\db\Exception
+     */
+    public function testSyncDataWithConnectionLost()
+    {
+        $this->app->db->createCommand('CREATE TABLE IF NOT EXISTS temp_synctest LIKE admin_user')->execute();
+        
+        $ctrl = new ProxyController('proxyctrl', $this->app);
+        $build = new ClientBuild($ctrl, [
+            'buildConfig' => ['tables' => ['temp_synctest' => ['name' => 'temp_synctest']]],
+        ]);
+        $table = new ClientTableMock($build, ['name' => 'temp_synctest']);
+    
+        try {
+            $table->syncData();
```

Comment:
you don't need try catch block as far as i know, you can just add 

$this->expectException('yii\db\Exception');

before the method which will thrown the exception.

https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

Reply:
This should be done thought, then we can merge